shared/naturalcrit/markdown.js

```diff
@@ -237,32 +237,45 @@ const superSubScripts = {
 const definitionLists = {
 	name  : 'definitionLists',
 	level : 'block',
-	start(src) { return src.match(/^.*?::.*/m)?.index; },  // Hint to Marked.js to stop and check for a match
+	start(src) { return src.match(/^.*?::.*\n\n/m)?.index; },  // Hint to Marked.js to stop and check for a match
 	tokenizer(src, tokens) {
-		const regex = /^([^\n]*?)::([^\n]*)(?:\n|$)/ym;
+		const regex = /^([^\n:]*?)::(.*)(?:\n|$)/ym;
 		let match;
-		let endIndex = 0;
-		const definitions = [];
+		const endIndex = src.match(`\n\n`)?.index + 2;
+		const allDefinitions = [];
+		let currentDefinition = [];
 		while (match = regex.exec(src)) {
-			definitions.push({
-				dt : this.lexer.inlineTokens(match[1].trim()),
-				dd : this.lexer.inlineTokens(match[2].trim())
-			});
-			endIndex = regex.lastIndex;
+			if(match[1].trim()?.length) {
+				if(currentDefinition?.dt?.length) {
+					allDefinitions.push(currentDefinition);
+					currentDefinition = [];
+				}
+				currentDefinition = {
+					dt : this.lexer.inlineTokens(match[1].trim()),
+					dd : []
+				};
+			}
+			currentDefinition.dd = currentDefinition.dd.concat(match[2].split('::').filter((item)=>item).map((s)=>this.lexer.inlineTokens(s.trim())));
+			if(!currentDefinition.dd?.length) {
+				currentDefinition.dd = [this.lexer.inlineTokens('')];
+			}
 		}
-		if(definitions.length) {
+		if(currentDefinition.hasOwnProperty('dt')) { allDefinitions.push(currentDefinition); }
+		if(allDefinitions.length) {
 			return {
-				type : 'definitionLists',
-				raw  : src.slice(0, endIndex),
-				definitions
+				type        : 'definitionLists',
+				raw         : src.slice(0, endIndex),
+				definitions : allDefinitions
 			};
 		}
 	},
 	renderer(token) {
-		return `<dl>${token.definitions.reduce((html, def)=>{
-			return `${html}<dt>${this.parser.parseInline(def.dt)}</dt>`
-			            + `<dd>${this.parser.parseInline(def.dd)}</dd>\n`;
-		}, '')}</dl>`;
+		let returnVal = `<dl>`;
+		token.definitions.forEach((def)=>{
+			const dds = def.dd.map((s)=>`<dd>${this.parser.parseInline(s)}</dd>`).join('\n');
```

Comment:
Can we change the behavior here slightly so all `<dd>`s are joined with `''` and newline `<dd>`s are also prepended with `'\n'`?

So the output looks like the markdown. i.e. preserve newlines so we can get.

![image](https://github.com/naturalcrit/homebrewery/assets/5878534/5d79c021-cc09-490f-b3d1-8a30e43e23b0)
vs
![image](https://github.com/naturalcrit/homebrewery/assets/5878534/63a33946-dbf0-425b-b567-f900292c6f37)
depending how the list is written.

For example:

```
Term :: def1 :: def2 :: def3
↓
<dl><dt>Term</dt><dd>def1</dd><dd>def2</dd><dd>def3</dd></dl>
```

and

```
Term
:: def1
:: def2
:: def3
↓
<dl><dt>Term</dt>
<dd>def1</dd>
<dd>def2</dd>
<dd>def3</dd></dl>
```

and I guess mixing inline and new lines would be:

```
Term :: def1
:: def2 :: def3
↓
<dl><dt>Term</dt><dd>def1</dd>
<dd>def2</dd><dd>def3</dd></dl>
```

This should result

Reply:
I've done this, though it causes.. funny behavior with the default styling which looks like it was designed solely for single term single definition, inline lists. It might be prudent to simply move that CSS from Blank to PHB/DMG if that was done purely for book emulation.

Reply:
Thabks! Ok. Ill take a look.